brain-marks/Services/MigrationService.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  MigrationService.swift
+//  brain-marks
+//
+//  Created by Jay on 12/13/22.
+//
+
+import CoreData
+import Foundation
+
+/// Controls the migration from Amplify to CoreData
+class MigrationService {
+    private let storageProvider = StorageProvider.shared
+    private var awsCategories: [AWSCategory] = []
+
+    private let amplifyDataStore = DataStoreManger.shared
+
+    private var managedObjectContext: NSManagedObjectContext
+
+    init() {
+        self.managedObjectContext = storageProvider.context
+    }
+
+    func performMigration() {
+        /// 1. Fetch categories
+        getCategories()
+        /// 2. Go through each category and get tweets
+        for awsCategory in awsCategories {
+            /// 3. Add category to core data
+            let categoryToAdd = makeCategoryCoreDataCompatible(category: awsCategory)
+            amplifyDataStore.fetchSavedTweets(for: awsCategory) { awsTweets in
+                for tweet in awsTweets ?? [] {
+                    let tweetToAdd = self.makeTweetCoreDataCompatible(tweet: tweet)
+                    categoryToAdd.addToTweets(tweetToAdd)
+                }
+            }
+            do {
+                try managedObjectContext.save()
+            } catch {
+                print("❌ MigrationService.performMigration() Error: \(error)")
+            }
+        }
+        UserDefaults.standard.set(true, forKey: "migrationToCoreDataRan")
+    }
+
+    public func checkIfMigrationShouldRun() -> Bool {
```

Comment:
```suggestion
    func checkIfMigrationShouldRun() -> Bool {
```

This doesn't need to be public, does it?

Reply:
Nope it doesn't